libmcount/dynamic.c

```diff
@@ -127,15 +129,25 @@ static int do_dynamic_update(struct symtabs *symtabs, char *patch_funcs)
 				stats.skipped++;
 				break;
 			case 0:
+				success = true;
 			default:
 				break;
 			}
 			stats.total++;
 		}
 
+		if (!success) {
+			nr_failed++;
+			func_name = name;
+		}
+
 		name = strtok(NULL, ";");
 	}
 
+	if (nr_failed)
+		pr_out("%s cannot be patched dynamically\n",
+		       nr_failed > 1 ? "some functions" : func_name);
+
```

Comment:
First, improving user visible message (more generally - user experience) is very important, and I appreciate you're working on this. :)

And for your concern, I think we can save first failing function name (or pattern) and then use it if it fails only once.  If stats.total is 0, we can count number of functions (or patterns).  You can add such fields to struct stats.